### PR TITLE
refactor(v2): init database package

### PR DESCRIPTION
**Pull request by adlerhurst**
This is the first package for the new internal package. We start with the `database` because it's imported by most other packages.

Part of https://github.com/zitadel/zitadel/issues/7639

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/v2/database/text_filter.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+func newTextFilter[T text](comp textCompare, t T) *TextFilter[T] {
+	return &TextFilter[T]{
+		Filter: Filter[textCompare, T]{
+			comp:  comp,
+			value: t,
+		},
+	}
+}
+
+func (f TextFilter[T]) Write(stmt *Statement, columnName string) {
```
Comment (livio-a) [inline internal/v2/database/text_filter.go:63]:
```suggestion
func (f *TextFilter[T]) Write(stmt *Statement, columnName string) {
```

File: internal/v2/database/text_filter.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		return
+	}
+	f.Filter.Write(stmt, columnName)
+}
+
+func (f *TextFilter[T]) writeCaseInsensitive(stmt *Statement, columnName string) {
+	stmt.WriteString("LOWER(")
+	stmt.WriteString(columnName)
+	stmt.WriteString(") ")
+	stmt.WriteString(f.comp.String())
```
Comment (livio-a) [inline internal/v2/database/text_filter.go:75]: this assumes that the value is always passed lower case, shouldn't we make sure here:
```suggestion
	stmt.WriteString(strings.ToLower(f.comp.String()))
```

Comment (livio-a) [reply internal/v2/database/text_filter.go:75]: current tests should be extended to reflect that as well

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/v2/database/text_filter.go (hunk 2 above)
Comment (adlerhurst) [reply internal/v2/database/text_filter.go:75]: added strings.ToLower to the constructors

**Review by livio-a — APPROVED** [review]

**Pull request merged**